src/cli/config.js

```diff
@@ -60,3 +60,26 @@ export const getSchemas = (schemaFilePath: string): [GraphQLSchema, Schema] => {
         return [schemaForValidation, schemaForTypeGeneration];
     }
 };
+
+/**
+ * Find the first item of the `config.generate` array where both:
+ * - no item of `exclude` matches
+ * - at least one item of `match` matches
+ */
+export const findApplicableConfig = (
+    path: string,
+    configs: Array<GenerateConfig> | GenerateConfig,
+): ?GenerateConfig => {
+    if (!Array.isArray(configs)) {
+        configs = [configs];
+    }
+    return configs.find((config) => {
```

Comment:
issue: Correct me if I'm wrong here, but this is what it looks like to me:
This returns the first matching path, and does not account for nesting paths / more specificity. For example, if `this/directory` happened to be listed before `this/directory/this-file.js`, it would ignore the more specific config.

Could you reduce over the array to find the most specific match? I had such a function in my commits.